core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -697,6 +703,8 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
       // threadIdx.z
       const dim3 block(m_rp.m_tile[0] * m_rp.m_tile[1], m_rp.m_tile[2],
                        m_rp.m_tile[3]);
+      KOKKOS_ASSERT(block.y > 0);
```

Comment:
why does this not check for block.x?

Reply:
I only added checks if we would divide by zero.